addons/mail_client_extension/controllers/main.py

```diff
@@ -0,0 +1,323 @@
+        return None
+
+    def _get_company_dict(self, company):
+        if not company:
+            return {'id': -1}
+
+        return {
+                    'id': company.id,
+                    'name': company.name,
+                    'phone': company.phone,
+                    'mobile': company.mobile,
+                    'email': company.email,
+                    'address': {
+                        'street': company.street,
+                        'city': company.city,
+                        'zip': company.zip,
+                        'country': company.country_id.name if company.country_id else ''
+                    },
+                    'website': company.website,
+                    'additionalInfo': json.loads(company.iap_enrich_info) if company.iap_enrich_info else {}
+                }
+
+    def _create_company_from_iap(self, domain):
+        iap_data = self._iap_enrich(domain)
+        if 'enrichment_info' in iap_data:
+            return None, iap_data['enrichment_info']
+
+        new_company_info = {
+            'is_company': True,
+            'name': iap_data.get("name"),
+            'street': iap_data.get("street_name"),
+            'city': iap_data.get("city"),
+            'zip': iap_data.get("postal_code"),
+            'phone': iap_data.get('phone_numbers', [''])[0],
+            'website': iap_data.get("domain"),
+            'email': iap_data.get('email', [''])[0]
+        }
+        if iap_data.get('country_code'):
+            country = request.env['res.country'].search([('code', '=', iap_data['country_code'].upper())])
+            if country:
+                new_company_info['country_id'] = country.id
+                if iap_data.get('state_code'):
+                    state = request.env['res.country.state'].search([
+                    ('code', '=', iap_data['state_code']),
+                    ('country_id', '=', country.id)
+                    ])
+                    if state:
+                        new_company_info['state_id'] = state.id
+
+        new_company_info['iap_enrich_info'] = json.dumps(iap_data)
+        new_company = request.env['res.partner'].with_user(request.env.user).create(new_company_info)
+
+        return new_company, {'type': 'company_created'}
+
+    @http.route('/mail_client_extension/partner/get', type="json", auth="outlook", csrf=False, cors="*")
+    def res_partner_get_by_email(self, email, name, **kwargs):
+        response = {
+            'authorized': True,
+        }
+
+        #compute the sender's domain
+        normalized_email = tools.email_normalize(email)
+        if not normalized_email:
+            response['error'] = 'Bad email.'
+            return response
+        sender_domain = normalized_email.split('@')[1]
+
+        # Search for the partner based on the email.
+        # If multiple are found, take the first one.
+        partner = request.env['res.partner'].with_user(request.env.user).search([('email', '=', email)], limit=1)
```

Comment:
Shouldn't this search use the normalised email? Or both e.g. `[('email', 'in', [normalized_email, email])]`?